lib/adapters/class-adapter.php

```diff
@@ -303,6 +310,111 @@ public function add_term_aggregation( string $label, string $term_field, string
 		);
 	}
 
+	/**
+	 * A callback to filter es args.
+	 * Adds phrase matching to the request args if it is enabled.
+	 *
+	 * @param array $es_args The request args to be filtered.
+	 * @return array The filtered request args.
+	 */
+	public function add_phrase_matching_to_es_args( array $es_args ): array {
+		if ( ! $this->get_enable_phrase_matching() ) {
+			return $es_args;
+		}
+
+		$search = get_query_var( 's' );
+
+		// Bail early if this isn't a search.
+		if ( empty( $search ) ) {
+			return $es_args;
+		}
+
+		// Break down the search string into the desired "phrase matched" parts.
+		// TODO: Add filter for phrase_match_delineator.
+		$phrase_match_delineator = '"';
+		preg_match_all( '/' . $phrase_match_delineator . '(.*?)' . $phrase_match_delineator . '/', $search, $matches );
+		if ( empty( $matches ) ) {
+			return $es_args;
+		}
+
+		// Get the search query without the "phrase matched" parts.
+		$unmatched = implode(
+			' ',
+			array_filter(
+				explode( ' ', preg_replace( '/' . $phrase_match_delineator . '.*?' . $phrase_match_delineator . '/', '', $search ) )
+			)
+		);
+
+		// Replace the main multi_match query with the "unmatched" string.
+		$es_arg = $this->find_multimatch_query( $es_args );
+		if ( ! empty( $es_arg['multi_match']['query'] ) ) {
+			if ( ! empty( $unmatched ) ) {
+				$es_arg['multi_match']['query'] = $unmatched;
+			} else {
+				unset( $es_arg );
+			}
+		}
+
+		/**
+		 * Filters the list of multimatch fields.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param array $default_multi_match_fields The array of multimatch fields with weighting to be included in phrase matching queries.
+		 */
+		$default_multi_match_fields = apply_filters(
+			'elasticsearch_extensions_phrase_match_multimatch_fields',
```

Comment:
Ideally, this could, or should, respect the active provider. One could easily hook into [VIP Search](https://github.com/Automattic/vip-go-mu-plugins-built/blob/93d4ba09a0fd308185867905a593edebe748290f/search/elasticpress/includes/classes/Indexable/Post/Post.php#L1459) and set a list of custom fields and forget, or not know, about this hook and have two, or more, separate fields for each query in a single request.

Reply:
The fields used here also have weights assigned to them. That to me assumes the default behavior has the same behavior. This can skew the results.

Reply:
These felt like sane defaults to me. This is another place I think where we could add an enhancement issue to hook into the provider and the already filtered values for these fields. Starting with a filterable set of fields to me, feels like a good first step.

Reply:
If that's documented, sure. If not, I'll realize, or not, later after activating it that my search results are not behaving as quite expected.

Reply:
these DSL changes are only applied if the delineators are found in the query, for example a search for `brown dog` will not include a multi_match phrase query in the DSL. On the other hand, a search for `"brown dog"` will return results that only have the exact phrase in one of these fields (weighted as they are above).

Reply:
Correct! I think I got this part. The part I'm bringing attention to is that it will return results using fields and weights differently from the ones set by the site. Will it matter? Probably not, but due to the impact in the score, if you don't know this behavior is there, it does matter.

Reply:
I got ya :) i definitely agree that this filter should be hooked into from the adapters and from there the fields should be passed through from the providers. I also think that this should be added as an issue instead of blocking this PR, since it will be obvious that the feature is responsible for manipulating the query when the feature is being used (i.e. when double quotes are added to the search and the feature is turned on).